src/python/nnfusion/__main__.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import sys, os, tarfile, shutil, logging
+from typing_extensions import runtime
+
+logging.basicConfig(level=os.environ.get("LOGLEVEL", "INFO"))
+this_folder = os.path.split(os.path.realpath(__file__))[0]
+nnf_bin=os.path.join(this_folder, "bin/nnfusion")
+nnf_dir=os.path.join(this_folder, "bin/")
+nnf_pkg = os.path.join(this_folder, "nnfusion.tar.gz")
+
+def check_pkg():
+    if os.path.exists(nnf_bin) and os.path.exists(nnf_pkg):
+        old_time = os.path.getctime(nnf_bin)
+        new_time = os.path.getctime(nnf_pkg) 
+        if new_time > old_time:
+            logging.info("Replacing old nnfusion cli.")
+            shutil.rmtree(nnf_dir)
+    else:
+        if os.path.exists(nnf_bin):
+            logging.info("No nnfusion cli found: Try to extract it from nnfusion.tar.gz.")
+
+
+def extract_pkg():
```

Comment:
May move extract_pkg to pip install instead of running it at each nnfusion compiling time.

This can occur write conflicts when run multiple nnfusion instances at same time (e.g., horovodrun/mpirun).

```
[0]<stderr>:Traceback (most recent call last):
[0]<stderr>: File "/usr/local/bin/nnfusion", line 8, in <module>
[0]<stderr>: sys.exit(main())
[0]<stderr>: File "/usr/local/lib/python3.6/dist-packages/nnfusion/__main__.py", line 58, in main
[0]<stderr>: extract_pkg()
[0]<stderr>: File "/usr/local/lib/python3.6/dist-packages/nnfusion/__main__.py", line 28, in extract_pkg
[0]<stderr>: tar.extractall(nnf_dir)
[0]<stderr>: File "/usr/lib/python3.6/tarfile.py", line 2012, in extractall
[0]<stderr>: numeric_owner=numeric_owner)
[0]<stderr>: File "/usr/lib/python3.6/tarfile.py", line 2054, in extract
[0]<stderr>: numeric_owner=numeric_owner)
[0]<stderr>: File "/usr/lib/python3.6/tarfile.py", line 2124, in _extract_member
[0]<stderr>: self.makefile(tarinfo, targetpath)
[0]<stderr>: File "/usr/lib/python3.6/tarfile.py", line 2165, in makefile
[0]<stderr>: with bltn_open(targetpath, "wb") as target:
[0]<stderr>:OSError: [Errno 26] Text file busy: '/usr/local/lib/python3.6/dist-packages/nnfusion/bin/nnfusion'
```